src/ucs/sys/netlink.c

```diff
@@ -285,10 +302,18 @@ int ucs_netlink_route_exists(int if_index, const struct sockaddr *sa_remote)
                                  NULL);
     }
 
-    info.if_index  = if_index;
-    info.sa_remote = sa_remote;
-    info.found     = 0;
+    info.if_index         = if_index;
+    info.sa_remote        = sa_remote;
+    info.found            = 0;
+    info.allow_default_gw = allow_default_gw;
+
     ucs_netlink_lookup_route(&info);
 
     return info.found;
 }
+
+int ucs_netlink_ethernet_device_route_exists(int if_index,
```

Comment:
Is there really anything "ethernet-specific" in this func?
+ see comment above "why not always..." maybe don't need 2 distinct funcs here?